src/pymor/algorithms/ml/vkoga.py

```diff
@@ -0,0 +1,426 @@
+        When adding :math:`x_{n+1}\in\mathbb{R}^d` as new center, we can compute
+        :math:`c_{n+1}\in\mathbb{R}^{n+1}` in the following way: Let us write
+
+        .. math::
+            c_{n+1} = \begin{bmatrix}
+                a_1 \\
+                a_2
+            \end{bmatrix}
+
+        for unknown coefficients :math:`a_1\in\mathbb{R}^n` and :math:`a_2\in\mathbb{R}`.
+        We then have (by splitting the right-hand side :math:`f_{n+1}` accordingly)
+
+        .. math::
+            \begin{align*}
+                K_na_1+Ba_2 &= f_1, \\
+                B^\top a_1+K_{nn}a_2 &= f_2.
+            \end{align*}
+
+        One can now solve for :math:`a_1` and :math:`a_2` in the following way:
+
+        .. math::
+            \begin{align*}
+                y_1 &= L_n^{-1} f_1,\\
+                y_2 &= L_{22}^{-1}(f_2-W^\top y_1),\\
+                a_1 &= L_n^{-\top}(y_1-Wy_2) = c_n - L_n^{-\top}Wy_2,\\
+                a_2 &= L_{22}^{-\top}y_2,
+            \end{align*}
+
+        where we observe that :math:`y_1` is given by the last iteration and therefore
+        the update of :math:`a_1` and the computation of :math:`a_2` are
+        relatively cheap.
+
+        Regarding the power function (measuring for :math:`x\in\mathbb{R}^d` how well
+        the current subspace can represent :math:`k(\,\cdot\,,x)`, i.e. the projection
+        error in the reproducing kernel Hilbert space):
+        The power function in matrix form is defined as
+
+        .. math::
+            \begin{align*}
+                P_n^2(x) = k(x,x) - k(x,X_n)k(X_n,X_n)^{-1}k(X_n,x).
+            \end{align*}
+
+        Since we would like to use the power function as selection criterion for centers
+        within the greedy iteration, we consider a scalar version by taking the trace:
+
+        .. math::
+            \begin{align*}
+                p_n^2(x) = \operatorname{trace}(P_n^2(x)).
+            \end{align*}
+
+        We are going to track this quantity evaluated at all training points during the
+        iterations of the greedy algorithm. In order to do so, we also maintain an array
+        storing the Newton basis evaluations of the current basis at the training points:
+
+        .. math::
+            \begin{align*}
+                V_{n,i} = L_n^{-1}k(X_n,y_i)
+            \end{align*}
+
+        for all training points :math:`y_i`.
+
+        Given the first center :math:`x_1`, we initialize the (squared) power function
+        values as
+
+        .. math::
+            \begin{align*}
+                p_n^2(y_i) = \operatorname{trace}(k(y_i,y_i)-V_{1,i}^\top V_{1,i}),
+            \end{align*}
+
+        where :math:`V_{1,i} = L_{22}^{-1}k(x_1,y_i)`.
+
+        The incremental updates of :math:`V_{n,i}` to :math:`V_{n+1,i}` and the power
+        function values is then performed in the following way:
+        Define :math:`\Delta_i=k(x_{n+1},y_i)-W^\top V_i` :math:`\Xi=L_{22}^{-1}\Delta_i`.
```

Comment:
This is not rendering correctly, in the documentation it looks like `\Delta_i = ... = L_{22}^{-1} \Delta_i`. Also, it should be `V_{n,i}` instead of `V_{i}`.